pennylane/transforms/hamiltonian_expand.py

```diff
@@ -97,21 +97,29 @@ def hamiltonian_expand(tape, group=True):
 
     hamiltonian = tape.measurements[0].obs
 
-    if not isinstance(hamiltonian, qml.Hamiltonian) or len(tape.measurements) > 1:
+    if (
+        not isinstance(hamiltonian, qml.Hamiltonian)
+        or len(tape.measurements) > 1
+        or tape.measurements[0].return_type != qml.measure.Expectation
+    ):
```

Comment:
The function expects an expval measurement but this was never checked before

Reply:
No, that would have not showed up in `tape.measurement`, but tapes with `return var(H)` would have been split into expval subtapes...